### PR TITLE
Feat/add utils dheeraj

### PR DESCRIPTION
**Pull request by dheerajsingh123**
Added truncateStringByCharacters, maskInputStringAndTruncate, smoothScrollToTop. Added comments to isValidEmail.

### CODE REVIEW

**Review by harshitpruthi — DISMISSED** [review]

File: src/dom/index.ts [hunk 1]
```diff
@@ -316,3 +316,24 @@ export function getOSName() {
...
+
+
+/**
+ * This function scrolls the page to the top.
+ * If window.scroll is available and works perfectly, this function uses the smooth scroll behaviour of window
+ * and scrolls with ease in animation. Else, It directly scrolls to top without animation in case of error with
+ * window.scroll
+ */
+export function smoothScrollToTop() {
+  if (typeof window !== 'undefined') {
```
Comment (vikaz596) [inline src/dom/index.ts:328]: use isEmpty method

File: src/string/index.ts [hunk 2]
```diff
@@ -171,14 +193,84 @@ export function capitalizeFirstLetter(str: string) {
...
+ * ```
+ *
+ * @category String Based Method
+ */
+export function truncateStringByCharacters(inputString: string = '', truncatedCharactersCount: number = 8) {
+  if (!isEmpty(inputString) && inputString.length >= truncatedCharactersCount) {
+    return inputString.substr(inputString.length - truncatedCharactersCount);
+
+  } else {
+    return '-';
```
Comment (vikaz596) [inline src/string/index.ts:225]: should return empty string

File: src/dom/index.ts [hunk 3]
```diff
@@ -316,3 +316,24 @@ export function getOSName() {
 
   return '';
 }
+
+
+/**
+ * This function scrolls the page to the top.
+ * If window.scroll is available and works perfectly, this function uses the smooth scroll behaviour of window
+ * and scrolls with ease in animation. Else, It directly scrolls to top without animation in case of error with
+ * window.scroll
```
Comment (vikaz596) [inline src/dom/index.ts:325]: category, example, etc are missing here.Docs wont be generated properly then.

File: src/dom/index.ts (hunk 1 above)
Comment (vikaz596) [reply src/dom/index.ts]: use isEMpty method instead of undefined

**Pull request merged**